rules/library.bzl

```diff
@@ -536,6 +540,62 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
                 opts = repr(v),
             ))
 
+    # Generate code for `.intentdefinition` source files and forward the original plist as data
+    for intent in intent_sources:
+        intent_name = "%s_%s_gen" % (name, intent)
+        testonly = kwargs.get("testonly", False)
+
+        # Avoid swift_intent_library and objc_intent_library because they wrap the generated code in a new module
+        #
+        # If other Swift sources are present, generate Swift intent code, otherwise use Obj-C.
+        # This mimics the behavior for INTENTS_CODEGEN_LANGUAGE="automatic" in Xcode.
+        if len(swift_sources) > 0:
+            apple_intent_library(
+                name = intent_name,
+                src = intent,
+                language = "Swift",
+                tags = _MANUAL,
+                testonly = testonly,
+            )
+
+            swift_sources.append(intent_name)
+            data.append(intent)
+
+        else:
+            intent_headers = "%s_hdrs" % intent_name
+            intent_sources = "%s_srcs" % intent_name
+
+            intent_public_header = paths.split_extension(intent)[0]
+
+            apple_intent_library(
+                name = intent_name,
+                src = intent,
+                header_name = intent_public_header,
+                language = "Objective-C",
+                tags = _MANUAL,
+                testonly = testonly,
+            )
+
+            native.filegroup(
+                name = intent_headers,
+                srcs = [intent_name],
+                output_group = "hdrs",
+                tags = _MANUAL,
+                testonly = testonly,
+            )
+
+            native.filegroup(
+                name = intent_sources,
```

Comment:
Why do we need a filegroup on this as opposed to just feeding the `objc_sources` the `intent_sources` as you did with the `swift_library` - same with the `intent_headers`.

If we need to have a `filegroup` for some reason to pass arbitrary files to `objc_library` - I'd suggest to just create one `filegroup(name=name+"objc_library_additonal_srcs", srcs= objc_library_additonal_srcs)` and add sources to the array as the builder progresses

Reply:
I wanted to use the `hdrs` and `srcs` outputs of the `apple_intent_library` separately, mostly so that the generated headers would be handled by any headermap logic down below.  Is there a more idiomatic way to do it that you can suggest?

Reply:
It looks like feeding `intent_name` to both `objc_hdrs` and `objc_sources` works.  Presumably `objc_library` is filtering them correctly either by name or via the provider?